packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,355 @@ void main()
+
+    } else {
+        info.miss = false;
+
+        if (length(rayDepth - hitCoordVS) < thickness * 0.05 //){
+                   || dot(normalize(dirVS), normalize(texture2D(normalSampler, uv).xyz)) > -0.085 ){
+            info.visibility = 0.0; // avoid displaying reflection when search level is low (auto intersection)
+        } else {
+            vec2 dCoordScreen = smoothstep(vec2(0.2), vec2(0.6), abs(vec2(0.5, 0.5) - uv)); // HermiteInterpolation
+            info.visibility = texture2D(textureSampler, uv).a // alpha value of the reflected scene position 
+                * abs(dot(normalize(hitCoordVS), normalize( texture2D(normalSampler, uv).xyz))) // to avoid artifacts due to missing information when refleted face is almost perpendicular to the camera view direction
+                
+                // reduce back face limits artifacts
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag + increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag - increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag + increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag - increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+              
+                * (1.0 - max (dot(-normalize(hitCoordVS), normalize(dirVS)), 0.0)) // to fade out the reflection as the reflected direction point to the camera's position (hit behind the camera)
+                * (1.0 - currSearch) // the reflection should be sharper when near from the starting point
+                * (1.0 - clamp (abs(hitCoordVS.z / distanceFade), 0.0, 1.0)) // to fade out the reflection near the distanceFade
+                * (1.0 - clamp (deltaDepth/maxTol, 0.0, 1.0)) // since the hit point is not always precisely found, we fade out the reflected color if we aren't precise enough 
+                * clamp(1.0 - (dCoordScreen.x + dCoordScreen.y), 0.0, 1.0); // to fade out the reflection near the edge of the screen
+        }
+
+        info.visibility = clamp(info.visibility * 2.0, 0.0, 1.0); // to compensate for previous lines to have strong enough reflections
+
+        #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+            info.visibilityBackup = 1.0 - info.visibility; // complementary reflectivityColor
+        #endif
+        info.coords = uv;
+    }
+
+    return info;
+}
+
+void main(void)
+{
+    #ifdef SSR_SUPPORTED
+
+    // *************** Get data from samplers ***************
+
+    vec4 original = texture2D(textureSampler, vUV);
+    vec3 spec = toLinearSpace(texture2D(specularSampler, vUV).rgb);
+
+    if (dot(spec, vec3(1.0)) <= 0.0){

```

Comment:
I tried to check for 0.12 and I have the same problem for standard and pbr materials: we miss some reflections / we loose reflection continuity and get ugly artifact
![bug](https://user-images.githubusercontent.com/44207625/180784074-58fa6016-4307-4a78-9014-88347380fb76.png)
s

Reply:
For standard materials you should not have the problem because the reflectivity values should be remapped from 0..1 to 0.04..1 before being saved in the reflectivity gbuffer (something like `new = old*0.96+0.04`, assuming the value is in linear space).

Regarding PBR, yes, testing for 0.04*3 is probably not the way to go because even a `metallic=0` material (dieletric) can have reflections depending on the roughness value:

**Roughness=0**:
![image](https://user-images.githubusercontent.com/4152247/180856927-04369d16-b084-44b2-85ed-1bf1c5cbc2c1.png)

**Roughness=1**:
![image](https://user-images.githubusercontent.com/4152247/180857311-98a6ef88-9d98-4dc6-8c2f-28f4aa904eb6.png)

The reflectivity of the material in both screenshots is the same, (0.04, 0.04, 0.04).

What bothers me by hard coding 0 is that we will have reflections computed for all texels when using PBR materials (because the lowest possible value is 0.04*3 which is > 0) even in cases where the reflection contributes nothing to the end result...

At least with the standard material we can decide where the reflections should occur thanks to the specular texture ((0,0,0) in the texture means no reflection and we can skip the computation). I wonder if we should not have the same possibility for PBR materials and have a texture that let the user choose what is reflective or not (wrt SSR)...

Let's keep the check as it is currently but we may need to revisit this question at some time.

Pinging @sebavan / @deltakosh so they are aware of the problem.